### PR TITLE
better Bluetooth Low Energy support

### PR DESCRIPTION
**Pull request by User65k**
Hi,

I added a few missing (ATT) bits for a better support of Bluetooth LE.
A lot of GATT stuff works out of the box by reusing EIR_Hdr instead of the own BTLE_AdvData.
The PPI_FieldHeader is switched with the generic PPI classes.

Greetings

### CODE REVIEW

**Comment by gpotter2** [comment]
Hi !

Thanks a lot for this PR ! We have a PEP8 test failing, see the trace below

```
scapy/
scapy/layers/bluetooth4LE.py:32:1: E302 expected 2 blank lines, found 1
scapy/layers/bluetooth4LE.py:125:5: E265 block comment should start with '# '
scapy/layers/bluetooth4LE.py:142:9: E265 block comment should start with '# '
scapy/layers/bluetooth4LE.py:196:1: E303 too many blank lines (3)
scapy/layers/bluetooth4LE.py:271:42: E261 at least two spaces before inline comment
scapy/layers/bluetooth4LE.py:271:43: E262 inline comment should start with '# '
```

File: scapy/layers/bluetooth4LE.py [hunk 1]
```diff
@@ -108,6 +123,11 @@ def swapbits(a):
                     state ^= lfsr_mask
         return struct.pack("<L", state)[:-1]
 
+    # def do_build(self):
+    #    #make sure post build is called
+    #    self.raw_packet_cache = None
+    #    super(Packet, self).do_build()
+
```
Comment (gpotter2) [inline scapy/layers/bluetooth4LE.py]: What is the point of this ?

**Comment by gpotter2** [comment]
Thanks ! Apart from a minor comment, all good to me.

**Comment by guedou** [comment]
Could you add some unit tests ? Thanks.

File: scapy/layers/bluetooth4LE.py (hunk 1 above)
Comment (User65k) [reply scapy/layers/bluetooth4LE.py]: I had some problems in the past with the CRC at the end. It is no longer needed, I will remove it.

**Pull request merged**